manticore/utils/config.py

```diff
@@ -0,0 +1,236 @@
+"""
+This file implements a configuration system.
+
+The config values and constant are gathered from three sources:
+
+    1. default values provided at time of definition
+    2. ini files (i.e. ./manticore.yml)
+    3. command line arguments
+
+in that order of priority.
+"""
+
+import ast
+import yaml
+import io
+import logging
+import os
+
+from itertools import product
+
+
+_groups = {}
+
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigError(Exception):
+    pass
+
+
+class _var:
+    def __init__(self, name: str='', default=None, description: str=None, defined: bool=True):
+        self.name = name
```

Comment:
idea;importance:3%. Make name and other private. `._name`
And if go that way. getter for value return default if None.

Reply:
name is used externally to the class though (i.e. in module functions)